CONTRIBUTING.md

```diff
@@ -1,3 +1,43 @@
-# How to become a contributor and submit your own code
+# Contributing Guidelines
 
-For instructions on how to contribute to the Kubernetes project please refer to the official [contributing guide](https://releases.k8s.io/HEAD/CONTRIBUTING.md)
+Read the following guide if you're interested in contributing to cluster-api.
+
+## Contributor License Agreements
+
+We'd love to accept your patches! Before we can take them, we have to jump a couple of legal hurdles.
+
+Please fill out either the individual or corporate Contributor License Agreement (CLA).
+
+  * If you are an individual writing original source code and you're sure you own the intellectual property, then you'll need to sign an [individual CLA](http://code.google.com/legal/individual-cla-v1.0.html).
+  * If you work for a company that wants to allow you to contribute your work, then you'll need to sign a [corporate CLA](http://code.google.com/legal/corporate-cla-v1.0.html).
+
+Follow either of the two links above to access the appropriate CLA and instructions for how to sign and return it. Once we receive it, we'll be able to accept your pull requests.
+
+***NOTE***: Only original source code from you and other people that have signed the CLA can be accepted into the repository.
+
+## Finding Things That Need Help
```

Comment:
Help might include actually deploying a cluster and see what works or not (i.e testing the code).

Reply:
That seems to be covered in the cluster-api-gcp README.